src/utils/string/StringUtils.js

```diff
@@ -398,6 +397,13 @@ const onOpenLink = (url) => {
   });
 };
 
+const isValidUrl = (url) => {
+  if (url && typeof url === 'string') {
+    return urlRegex.test(url);
+  }
+  return false;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `isValidUrl` function correctly checks if the input is a string and if it matches the `urlRegex`. However, the `test()` method of a RegExp object in JavaScript is stateful when used with global flag 'g'. If you use it multiple times on the same regex object, it will start from where it left off in the previous match instead of starting from the beginning. To avoid this, you can create a new RegExp object each time you want to test a string.
```diff
- return urlRegex.test(url);
+ return new RegExp(urlRegex).test(url);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
need to check if url is string first